src/windowgroup.cpp

```diff
@@ -64,7 +64,12 @@ void WindowGroup::setupWindow(QWindow *window, Maliit::Position position)
             window->setFlags(Qt::Window
                              | Qt::FramelessWindowHint
                              | Qt::WindowStaysOnTopHint
-                             | Qt::WindowDoesNotAcceptFocus);
+                             | Qt::WindowDoesNotAcceptFocus
+                             /* Mir uses a special window type for input method
+                              * panel, which is not handled by Qt::WindowType
+                              * natively, so add the magic value here
+                              */
+                             | static_cast<Qt::WindowType>(0x00000080));
```

Comment:
Not much harm, but as commented on PR 101, this could be avoided here if Mir had its own platform class that implemented setupInputPanel().

Reply:
I don't see anything in #101 relating to Mir (or anything else) having a platform class. That also seems like an excessive thing to do for one integer flag on the window. Mir 1.8 also supports both mirclient and wayland protocols, while Mir 2.x only supports wayland (although, I'm not sure if it has input-panel-shell, input-method, text-input support yet).